src/stravalib/model.py

```diff
@@ -220,6 +231,148 @@ def check_valid_location(
+
+    @classmethod
+    @abstractmethod
+    def get_core_schema(cls) -> Mapping[str, Any]: ...
+
+    @classmethod
+    def __get_pydantic_core_schema__(
+        cls,
+        _source_type: BaseType,
+        _handler: GetCoreSchemaHandler,
+    ) -> core_schema.CoreSchema:
+        from_schema = core_schema.chain_schema(
+            [
+                cls.get_core_schema(),
+                core_schema.no_info_plain_validator_function(cls.validate),
+            ]
+        )
+
+        return core_schema.json_or_python_schema(
+            json_schema=from_schema,
+            python_schema=core_schema.union_schema(
+                [core_schema.is_instance_schema(cls._type), from_schema]
+            ),
+            serialization=core_schema.plain_serializer_function_ser_schema(
+                lambda v: v
+            ),
+        )
+
+    @classmethod
+    def __get_pydantic_json_schema__(
+        cls,
+        _core_schema: core_schema.CoreSchema,
+        handler: GetJsonSchemaHandler,
+    ) -> JsonSchemaValue:
+        return handler(cls.get_core_schema())
+
+
+class _CustomIntAnnotation(_CustomTypeAnnotation[int, Any], ABC):
+    @classmethod
+    def get_core_schema(cls) -> Mapping[str, Any]:
+        return core_schema.int_schema()
+
+
+class _CustomFloatAnnotation(_CustomTypeAnnotation[float, Any], ABC):
+    @classmethod
+    def get_core_schema(cls) -> Mapping[str, Any]:
+        return core_schema.float_schema()
+
+
+class _CustomStrAnnotation(_CustomTypeAnnotation[str, Any], ABC):
+    @classmethod
+    def get_core_schema(cls) -> Mapping[str, Any]:
+        return core_schema.str_schema()
+
+
+class Duration(int):
+    def timedelta(self) -> timedelta:
+        return timedelta(seconds=self)
+
+
+class _DurationAnnotation(_CustomIntAnnotation):
+    _type = Duration
+
+    @classmethod
+    def validate(cls, value: int) -> Duration:
+        return Duration(value)
+
+
+class Timezone(str):
+    def timezone(
+        self,
+    ) -> (
+        pytz._UTCclass
+        | pytz.tzinfo.StaticTzInfo
+        | pytz.tzinfo.DstTzInfo
+        | None
+    ):
+        if " " in self:
+            # (GMT-08:00) America/Los_Angeles
+            tzname = self.split(" ", 1)[1]
+        else:
+            # America/Los_Angeles
+            tzname = self
+        try:
+            return pytz.timezone(tzname)
+        except UnknownTimeZoneError as e:
+            LOGGER.warning(
+                f"Encountered unknown time zone {tzname}, returning None"
+            )
+            return None
+
+
+class _TimezoneAnnotation(_CustomStrAnnotation):
+    _type = Timezone
+
+    @classmethod
+    def validate(cls, value: str) -> Timezone:
+        return Timezone(value)
+
+
+class Distance(_Quantity):
+    unit = "meters"
+
+
+class _DistanceAnnotation(_CustomFloatAnnotation):
+    _type = Distance
+
+    @classmethod
+    def validate(cls, value: float) -> Distance:
+        return Distance(value)
+
+
+class Velocity(_Quantity):
```

Comment:
Let's add a short description here that directs users to unit_helper. And we could also include a small example  using unit_helper. i'm just starting to understand how you pulled this all together - bare with me 😸